codegen/smithy-ruby-codegen-test-utils/src/main/java/software/amazon/smithy/ruby/codegen/integrations/WhiteLabelTestIntegration.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.ruby.codegen.integrations;
+
+import java.util.List;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.ruby.codegen.GenerationContext;
+import software.amazon.smithy.ruby.codegen.RubyIntegration;
+import software.amazon.smithy.ruby.codegen.RubyRuntimePlugin;
+import software.amazon.smithy.ruby.codegen.config.ClientConfig;
+import software.amazon.smithy.ruby.codegen.middleware.Middleware;
+import software.amazon.smithy.ruby.codegen.middleware.MiddlewareBuilder;
+import software.amazon.smithy.ruby.codegen.middleware.MiddlewareStackStep;
+
+public class WhiteLabelTestIntegration implements RubyIntegration {
+
+    @Override
+    public boolean includeFor(ServiceShape service, Model model) {
```

Comment:
This is saying apply the integration only for the whitelabel service.  The cool part of this is that we can apply integrations in a very flexible way - ie, the Checksum RubyIntegration only applies to services that have the httpChecksum trait somewhere in their model.  

I dont see another way to do this.  RubyIntegrations are Java SPI, so they are loaded if they are on the class path.  You need to have some runtime method of deciding if the RubyIntegration should be applied during each service generation.  

Do you know another pattern we should use instead?

Reply:
Are Integrations similar to this Smithy plugin concept? I was hoping for a way to just register integrations per service instead of loading all of them and then having a method boolean check if it applies or not. https://smithy.io/1.0/guides/building-models/build-config.html#plugins